influence_benchmark/RL/SFT.py

```diff
@@ -121,6 +123,8 @@ def formatting_prompts_func(example):
         data_collator=collator,
         max_seq_length=args.max_seq_length,
     )
+    # Remove the columns that are not needed or it will cause errors, as training will try to cast these strings to tensors
+    trainer.train_dataset = trainer.train_dataset.remove_columns(["text", "messages"])
```

Comment:
I'm a bit confused about why this would show up now?

Reply:
In order to access the `num_messages_to_ignore` for each training example at Collator time, I needed to add the `remove_unused_columns=False` arg to SFT, which by default causes the errors described in this comment

Reply:
(Figuring out how to do this was the painful part)